src/Panel.jsx

```diff
@@ -23,22 +35,29 @@ var Panel = React.createClass({
     };
   },
 
-  handleSelect: function (e) {
-    if (this.props.onSelect) {
-      this._isChanging = true;
-      this.props.onSelect(this.props.eventKey);
-      this._isChanging = false;
+  handleSelect: function(e){
+    if(this.props.onSelect) {
+      this.props.onSelect(e, this.props.eventKey)
+    } else {
+      this.handleClickEvent(e);
+      this.handleToggle();
     }
+  },
 
+  handleClickEvent: function(e){
     e.preventDefault();
+  },
 
-    this.setState({
-      expanded: !this.state.expanded
-    });
+  handleToggle: function(){
```

Comment:
Thanks makes sense to be the method name in the mixin. Ideally i'd like to figure out how to have the mixin basically handle this toggling itself.